packages/fxa-settings/src/pages/Signin/utils.ts

```diff
@@ -200,6 +200,12 @@ export async function handleNavigation(navigationOptions: NavigationOptions) {
     integration.isSync() ||
     integration.isFirefoxClientServiceRelay() ||
     integration.isFirefoxClientServiceAiMode();
+  const wantsTwoStepAuthentication = isOAuth && 'wantsTwoStepAuthentication' in integration
```

Comment:
I believe instead of these checks as well as the `isOAuth && !isOAuthNativeIntegration(integration)` check you have below, you can just check for `isOAuthWebIntegration`.

Also, I think you're going to need a tweak [here](https://github.com/mozilla/fxa/blob/main/packages/fxa-settings/src/pages/Signin/index.tsx#L238). I haven't tested your PR locally yet but I suspect these users are getting the signin email code as well but no way to enter it.